Add grot guide docs (#820)
grafana/writers-toolkit: docs/sources/write/shortcodes/index.md

{{< /collapse >}}

> ## Guide

Reviewer: Can we move this to be alphabetical in the page table of contents.

Author: Sure, if we don't move it to its own page I'll reorder.

> The `guide` shortcode embeds a Grot guide into the docs that people can follow to get help and guidance on topics and direction to documentation resources.

Reviewer: ```suggestion
The `guide` shortcode embeds a Grot guide into the documentation that people can follow to get help and guidance on topics and direction to documentation resources.
```

```markdown

> {{</* guide name="instrument" title="title" text="text" button="button" */>}}

Reviewer: https://grafana.com/docs/writers-toolkit/write/style-guide/write-for-developers/#placeholder-variables

```suggestion
{{</* guide name="<NAME>" title="<TITLE>" text="<TEXT>" button="<BUTTON>" */>}}
```

```

> The website looks up guide Markdown files from the website repository in the `content/guides` folder. The shortcode accepts the following parameters:

Reviewer: ```suggestion
The website looks up guide Markdown files from the website repository in the `content/guides` directory.
The shortcode accepts the following parameters:
```

| Parameter | Description                                                                 | Required |
| --------- | --------------------------------------------------------------------------- | -------- |
| `name`    | The name of the guide Markdown file in the website `content/guides` folder. | yes      |
| `title`   | A custom guide title to override the default guide title.                   | no       |
| `text`    | Custom body text to override the default guide text.                        | no       |
| `button`  | Custom button copy to override the default guide button copy.               | no       |

> If you don't set custom title, text, and button values, the guide introduction screen will use the following default text:

Reviewer: ```suggestion
If you don't set custom title, text, and button values, the guide introduction screen uses the following default text:
```

> ![Grot guide introduction screen with default copy](https://grafana.com/media/docs/writers-toolkit/grot-guide.png)

Reviewer: For accessibility, should we replace the image with the actual text for each parameter?

Author: It's also to give a visual representation of the grot guide.

      screen_id: instrument
```

### Screens

Define subsequent screens as a list of objects under a `screens` field:

```yaml
screens:
  - type: question
    id: ...

  - type: result
    id: ...
```

#### Question screen

A question screen is a branch node in a decision tree and presents one or many options to further screens. A question screen has the following structure:

```yaml
- type: question
    id: instrument
    title: What programming language do you want to instrument?
    options:
      - text: JVM (Java, Scala, Kotlin)
        screen_id: jvm
      - text: .Net
        screen_id: dotnet
      - text: Node.js
        screen_id: nodejs
```

### Result screen

A result screen is a leaf node in a decision tree and terminates with one or many links. A question screen has the following structure:

```yaml
  - type: result
    id: beyla
    title: Grafana Beyla
    body: |
      Grafana Beyla is a Linux eBPF kernel module application to auto-instrument applications without modifying them by monitoring an executable or port.
    links:
      - type: docs
        title: Grafana Beyla
        link_text: Visit docs page
        href: /docs/beyla/latest/
```

> 

Reviewer: What do you think about creating a new Grot guides page? The shortcodes reference page is already quite long and this section pretty large.

Author: We can do that, would you like it under `write/grot-guide.md`?

Reviewer: That makes sense to me :)
